examples/cpp/flow_control/PublisherApp.cpp

```diff
@@ -0,0 +1,219 @@
+
+using namespace eprosima::fastdds::dds;
+using namespace eprosima::fastdds::rtps;
+
+namespace eprosima {
+namespace fastdds {
+namespace examples {
+namespace flow_control {
+
+PublisherApp::PublisherApp(
+        const CLIParser::flow_control_config& config)
+    : participant_(nullptr)
+    , fast_publisher_(nullptr)
+    , slow_publisher_(nullptr)
+    , topic_(nullptr)
+    , fast_writer_(nullptr)
+    , slow_writer_(nullptr)
+    , type_(new FlowControlPubSubType())
+    , matched_(0)
+    , samples_(config.samples)
+    , stop_(false)
+{
+    // Create Participant
+    DomainParticipantQos pqos = PARTICIPANT_QOS_DEFAULT;
+    pqos.wire_protocol().builtin.discovery_config.leaseDuration = eprosima::fastdds::c_TimeInfinite;
+    pqos.name("Participant_publisher");
+
+    // This controller allows 300kb per second.
+    auto slow_flow_controller_descriptor = std::make_shared<eprosima::fastdds::rtps::FlowControllerDescriptor>();
+    slow_flow_controller_descriptor->name = "slow_flow_controller_descriptor";
+    slow_flow_controller_descriptor->max_bytes_per_period = config.max_bytes_per_period;
+    slow_flow_controller_descriptor->period_ms = config.period;
+    slow_flow_controller_descriptor->scheduler = config.scheduler;
+    pqos.flow_controllers().push_back(slow_flow_controller_descriptor);
+
+    participant_ = DomainParticipantFactory::get_instance()->create_participant(0, pqos);
+    if (participant_ == nullptr)
+    {
+        throw std::runtime_error("Participant initialization failed");
+    }
+
+    //Register the type
+    type_.register_type(participant_);
+
+    // Create fast Publisher, which has no controller of its own.
+    fast_publisher_ = participant_->create_publisher(PUBLISHER_QOS_DEFAULT, nullptr, StatusMask::none());
+    if (fast_publisher_ == nullptr)
+    {
+        throw std::runtime_error("Fast Publisher initialization failed");
+    }
+
+    // Create Topic
+    topic_ = participant_->create_topic("flow_control_topic", type_.get_type_name(), TOPIC_QOS_DEFAULT);
+
+    if (topic_ == nullptr)
+    {
+        throw std::runtime_error("Topic initialization failed");
+    }
+
+    // Create fast DataWriter
+    DataWriterQos wfqos = DATAWRITER_QOS_DEFAULT;
+    wfqos.publish_mode().kind = ASYNCHRONOUS_PUBLISH_MODE;
+    wfqos.data_sharing().off();
+
+    fast_writer_ = fast_publisher_->create_datawriter(topic_, wfqos, this, StatusMask::all());
+    if (fast_writer_ == nullptr)
+    {
+        throw std::runtime_error("Fast DataWriter initialization failed");
+    }
+    std::cout << "Fast publisher created, waiting for Subscribers." << std::endl;
+
+    // Create slow Publisher, with its own controller
+    slow_publisher_ = participant_->create_publisher(PUBLISHER_QOS_DEFAULT, nullptr, StatusMask::none());
+    if (slow_publisher_ == nullptr)
+    {
+        throw std::runtime_error("Slow Publisher initialization failed");
+    }
+
+    // Create slow DataWriter
+    DataWriterQos wsqos = DATAWRITER_QOS_DEFAULT;
```

Comment:
Get the qos from a default XML profile. That way, on the XML we can set appropriate Qos so the test is consistently passing without leaking test details into the example's code. This applies also for participant Qos, the other writer, and the reader

- Durability as TRANSIENT_LOCAL
- Reliability as RELIABLE
- History depth to a higher number